BufferGame/HistoricalData.cs

```diff
@@ -0,0 +1,147 @@
+﻿using static BufferGame.GlobalData;
+
+namespace BufferGame
+{
+    public class HistoricalData
+    {
+        public Logger Logger { get; set; }
+        public List<int> Values { get; set; }
+        public List<Description> ListDescription { get; set; }
+        public event Action<List<HistoricalProperty>> OnHistoricalDataReady;
+
+        public HistoricalData(Logger logger)
+        {
+            Logger = logger;
+            Values = new List<int>();
+            ListDescription = new List<Description>();
+        }
+
+
+        public void StoreDataFromBufferNew(CollectionDescription collectionDescription)
+        {
+            // provera da li je odgovarajuc dataset sa kodovima
+            if (ValidateData(collectionDescription) == false)
+            {
+                Logger.Log($"Dataset and code not valid\n");
+                return;
+            }
+            foreach (var description in ListDescription)
```

Comment:
foreach (var description in ListDescription)
        {
            if (description.Dataset == collectionDescription.Dataset)
            {
                ProcessExistingDataset(description, collectionDescription.BufferPropertyCollection);
                return;
            }
        }

        ProcessNewDataset(collectionDescription);
    }
    
    isto razdvojiti u manje metode, lakse je posle snaci se, a ako stavimo i logove u te iste metode lakse mozemo pronaci gde je bug